test/uk/gov/hmrc/merchandiseinbaggage/CoreTestData.scala

```diff
@@ -193,4 +197,51 @@ trait CoreTestData {
       case Import => GoodsEntries(completedImportGoods)
       case Export => GoodsEntries(completedExportGoods)
     }
+
+  def generateDeclarationConfirmationPage(
+    decType: DeclarationType,
+    purchaseAmount: Long)(implicit app: Application, message: Messages, appConfig: AppConfig): String = {
+    val layout = app.injector.instanceOf[Layout]
+    val link = app.injector.instanceOf[uk.gov.hmrc.merchandiseinbaggage.views.html.components.link]
+    val dummyAmount = AmountInPence(0)
+
+    def totalCalculationResult: TotalCalculationResult =
+      TotalCalculationResult(
+        CalculationResults(
+          Seq(CalculationResult(
+            ImportGoods(
+              CategoryQuantityOfGoods("sock", "1"),
+              GoodsVatRates.Twenty,
+              YesNoDontKnow.Yes,
+              PurchaseDetails(purchaseAmount.toString, Currency("GBP", "title.british_pounds_gbp", None, List.empty[String]))
+            ),
+            AmountInPence(purchaseAmount),
+            dummyAmount,
+            dummyAmount,
+            None
+          ))),
+        AmountInPence(purchaseAmount),
+        dummyAmount,
+        dummyAmount,
+        dummyAmount
+      )
+
+    val sessionId = SessionId()
+    val id = DeclarationId("456")
+    val created = LocalDateTime.now.withSecond(0).withNano(0)
+
+    val journey: DeclarationJourney = completedDeclarationJourney
+      .copy(sessionId = sessionId, declarationType = decType, createdAt = created, declarationId = id)
+
+    val persistedDeclaration = journey.declarationIfRequiredAndComplete.map { declaration =>
+      if (decType == DeclarationType.Import) declaration.copy(maybeTotalCalculationResult = Some(totalCalculationResult))
+      else declaration
+    }
+
+    val declarationConfirmationView = new DeclarationConfirmationView(layout, null, link)
```

Comment:
i think you can get this View from Injector directly without having to create an instance like this ?